uw-frame-components/portal/notifications/services.js

```diff
@@ -4,64 +4,186 @@ define(['angular', 'jquery'], function(angular, $) {
-    var dismissedIds = [];
 
     /////////////////////
     // SERVICE METHODS //
     /////////////////////
+    
     /**
      * Get all notifications at the given URL (as set in app-config.js or override.js)
      * @returns {*} A promise which returns an object containing notifications arrays
      */
     var getAllNotifications = function() {
-      return $q.all([$http.get(SERVICE_LOC.notificationsURL, {cache : true}), getDismissedNotificationIds()])
-          .then(function(result) {
-            return sortNotifications(result[0].data.notifications, result[1]);
-          },
-          function(reason) {
-            miscService.redirectUser(reason.status, 'notifications json feed call');
-          }
-        );
+      return $http.get(SERVICE_LOC.notificationsURL, {cache : true}).then(
+        function(result) {
+          return separateNotificationsByDismissal(result.data.notifications).then(
+            function(results){
+              return results;
+            },function(reason){
+              $log.warn("Error Retrieving all notifications");
+            }
+          );
+        },
+        function(reason) {
+          miscService.redirectUser(reason.status, 'notifications json feed call');
+        }
+      );
     };
-
+    
     /**
-     * Asynchronously fetch notifications and portal groups, then filter notifications by group. Called if group
-     * filtering is enabled.
+     * Gets notifications and filters them by group and via data if dataURL is
+     * present.
      * @returns {*} A promise which returns an object containing notifications arrays
      */
-    var getNotificationsByGroups = function() {
-      // If $q already happened, just return the promise
-      if(filteredNotificationPromise) {
-        return filteredNotificationPromise;
-      }
+    var getFilteredNotifications = function() {
+      var notifications = {
+        'dismissed': [],
+        'notDismissed': []
+      };
+      return $http.get(SERVICE_LOC.notificationsURL, {cache : true}).then(function(allNotifications){
+              return filterNotificationsByGroup(allNotifications.data.notifications);
+         }).then(function(filteredNotificationsByGroup){
+              return separateNotificationsByDismissal(filteredNotificationsByGroup);
+         }).then(function (notificationsBydismissal){
+              notifications.dismissed = notificationsBydismissal.dismissed;
+              return filterNotificationsByData(notificationsBydismissal.notDismissed);
+         }).then(function(notificationsByData){
+              notifications.notDismissed = notificationsByData;
+              return notifications;
+         }).catch(function(reason){
+             $log.warn("Error in retrieving notifications");
+             return [];
+         });
+    };
+    
+    /**
+     * Filter the array of notifications based on the groups that were passed in
+     * @param arrayOfNotifications : an array of notifications
```

Comment:
Could the JSDoc comments include param and return types? ([reference](http://usejsdoc.org/tags-param.html))